libs/slam/async_slam/async_slam_worker.cpp

```
@@ -0,0 +1,282 @@
+
+bool AsyncSlam::GetLatestProcessingImages_worker(FrameId frame_id, Images& current_images) {
+  {
+    std::lock_guard image_guard(processing_images_mutex_);
+    current_images = processing_images_;
+  }
+  const auto current_image =
+      std::find_if(current_images.begin(), current_images.end(), [](const auto& image) { return image != nullptr; });
+  return current_image != current_images.end() && ((*current_image)->get_image_meta().frame_id == frame_id);
+}
+
+void AsyncSlam::RunLoopClosureAndOptimization_worker(FrameId frame_id, uint64_t timestamp_ns,
```

Clang (14.0.6)

[warning] function 'RunLoopClosureAndOptimization_worker' has cognitive complexity of 29 (threshold 25)

(readability-function-cognitive-complexity)